tests/test_interrupted_sync.py

```diff
@@ -0,0 +1,268 @@
+                        if record.get('action') == 'upsert'
+                        and record.get('data', {}).get('owner_resource') == 'shop']
+
+                replication_key = next(iter(expected_replication_keys[stream]))
+                first_bookmark_stream = first_sync_state.get('bookmarks', {}).get(stream, {})
+                first_bookmark_value = first_bookmark_stream.get(replication_key)
+                resuming_bookmark_stream = resuming_sync_state.get('bookmarks', {}).get(stream, {})
+                resuming_bookmark_value = resuming_bookmark_stream.get(replication_key)
+                resuming_bookmark_value_utc = self.convert_state_to_utc(resuming_bookmark_value)
+
+                if stream in new_state['bookmarks'].keys():
+                    simulated_bookmark = new_state['bookmarks'][stream]
+                    simulated_bookmark_value = simulated_bookmark[replication_key]
+
+                youngest_first_sync_date = max(
+                    self.parse_date(record.get(replication_key))
+                    for record in first_sync_messages)
+
+                # verify the syncs sets a bookmark of the expected form
+                self.assertIsNotNone(first_bookmark_value)
+                self.assertTrue(self.is_expected_date_format(first_bookmark_value))
+                self.assertIsNotNone(resuming_bookmark_value)
+                self.assertTrue(self.is_expected_date_format(resuming_bookmark_value))
+
+                # verify the resuming bookmark is greater than 1st sync bookmark
+                # This is the expected behaviour for shopify as they are using date windowing
+                # TDL-17096 : Resuming bookmark value is getting assigned from execution time
+                # rather than the actual bookmark time for some streams.
+                # TODO this is not the case for the transactions stream, they are equal.
+                if stream == 'transactions':
+                    self.assertEqual(resuming_bookmark_value, first_bookmark_value)
+                else:
+                    self.assertGreater(resuming_bookmark_value, first_bookmark_value)
+
+                # if metafields stream has no 'shop' messages resuming_sync_messages can be empty
+                if resuming_sync_messages:
+                    actual_oldest_resuming_replication_date = min(
+                        self.parse_date(record.get(replication_key))
+                        for record in resuming_sync_messages)
+
+                # verify oldest record from resuming sync respects bookmark from previous sync
+                if stream in new_state['bookmarks'].keys() and resuming_sync_messages:
+                    self.assertEqual(actual_oldest_resuming_replication_date,
+                                     self.parse_date(simulated_bookmark_value),
+                                     msg="Oldest resuming sync record not respecting bookmark")
```

Comment:
Is there a reason we shouldn't combine these?

| `stream in new_state['bookmarks'].keys()` | `resuming_sync_messages` exists | Result          | `actual_oldest_resuming_replication_date` defined? |
|-------------------------------------------|---------------------------------|-----------------|----------------------------------------------------|
| True                                      | True                            | run the assert  | Yes                                                |
| True                                      | False                           | skip the assert | No                                                 |
| False                                     | True                            | skip the assert | Yes                                                |
| False                                     | False                           | skip the assert | No                                                 |

The only time we run the assert and have `actual_oldest_resuming_replication_date` defined is the first line

Reply:
Combined